ql/interestrate.hpp

```diff
@@ -185,6 +185,15 @@ namespace QuantLib {
         Real freq_;
     };
 
+    // helper to avoid relying on implicit conversion to Rate
+    inline Rate operator-(const InterestRate& a, const InterestRate& b) {
```

Comment:
Hmm, I didn't realize one could add or subtract `InterestRate` instances.  I would probably remove these operators and write an explicit call to `rate` at the point of use, to avoid adding or subtracting rates with different conventions.

Reply:
We tend to agree on that point. However, in that case I would remove the implicit conversion operator in the `InterestRate` class to `Rate` (i.e. `Real`) altogether. Otherwise such operations can be added without developers noticing. 

One example where this is used is in [ql/experimental/processes/extendedblackscholesprocess.cpp](https://github.com/lballabio/QuantLib/blob/master/ql/experimental/processes/extendedblackscholesprocess.cpp#L39-L40), but there are 15 uses in total throughout the library. 

What do you think?

Reply:
I don't disagree, but removing the implicit conversion would break backward compatibility.

Removing these operators at least would help us find and avoid implicit conversions in subtractions and additions, which are more at risk of inconsistency because they involve two different rates.

Reply:
That's just pushed - we changed it at the point of use to an explicit call to `rate` and removed those operators.